### PR TITLE
Fix catchall route check

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

regex * is not valid for URI, we should check .*

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
cc @wulianglongrd @ramaraochavali

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 1]
```diff
@@ -1563,7 +1535,7 @@ func isCatchAllRoute(r *route.Route) bool {
 	case *route.RouteMatch_PathSeparatedPrefix:
 		catchall = ir.PathSeparatedPrefix == "/"
 	case *route.RouteMatch_SafeRegex:
-		catchall = ir.SafeRegex.GetRegex() == "*"
```
Comment (zirain) [inline pilot/pkg/networking/core/v1alpha3/route/route.go:1566]: this due to envoy use RE2?

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/route/route.go:1566]: yes, RE2 treat it with error

**Review by howardjohn — COMMENTED** [review]
lgtm, double check with @ramaraochavali though

**Comment by ramaraochavali** [comment]
Can you provide more context? Is this not working today if some creates a VS with regex match `*` ?

**Comment by ramaraochavali** [comment]
LGTM

**Review by ramaraochavali — APPROVED** [review]

File: pilot/pkg/networking/core/v1alpha3/route/route.go [hunk 2]
```diff
@@ -1563,7 +1535,7 @@ func isCatchAllRoute(r *route.Route) bool {
...
+		catchall = ir.SafeRegex.GetRegex() == ".*"
 	}
 	// A Match is catch all if and only if it has no header/query param match
 	// and URI has a prefix / or regex *.
```
Comment (wulianglongrd) [inline pilot/pkg/networking/core/v1alpha3/route/route.go:1541]: nit: the comment looks incorrect, it should be `.*`

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/route/route.go:1541]: ack

**Pull request merged**